python/http_ece/__init__.py

```diff
@@ -1,76 +1,143 @@
-import os, struct, functools
+import functools
+import os
+import struct
+
+from cryptography.exceptions import InvalidTag
+from cryptography.hazmat.backends import default_backend
 from cryptography.hazmat.primitives import hashes
 from cryptography.hazmat.primitives.kdf.hkdf import HKDF
-from cryptography.hazmat.backends import default_backend
 from cryptography.hazmat.primitives.ciphers import (
     Cipher, algorithms, modes
 )
-import pyelliptic
+from pyelliptic import ecc
 
 keys = {}
 labels = {}
 
-def deriveKey(mode, salt, key=None, dh=None, keyid=None, authSecret=None, padSize=2):
-    def buildInfo(base, context):
-        return b"Content-Encoding: " + base + b"\0" + context
+MAX_BUFFER_SIZE = pow(2, 36) - 31
```

Comment:
You can trim this down to 2^32-1.  I doubt that anyone using the old stuff will care.